### PR TITLE
[Feature] Add metaschema model definition, list and detail endpoints [OSF-7457]

### PR DESCRIPTION
**Pull request by fabmiz**
#### Purpose

Add documentation for metaschemas endpoints.

#### Changes

- Creates metaschemas model definition.
- Adds metaschemas detail endpoint (/metaschemas/<metaschema_id>/ ) documentation (detail.yaml).
- Adds metaschemas list endpoint (/metaschemas/) documentation (list.yaml).

#### Ticket
- [OSF-7457](https://openscience.atlassian.net/browse/OSF-7457)

### CODE REVIEW

File: swagger-spec/metaschemas/list.yaml [hunk 1]
```diff
@@ -1 +1,382 @@
...
+    for [pagination](#Introduction_pagination).
+
+
+    This request should never return an error.
+
+  tags:
+    - Metaschemas
+  operationId: metaschemas_list
+  x-response-schema: Metaschema
+  responses:
```
Comment (caseyrollins) [inline swagger-spec/metaschemas/list.yaml:29]: In the interest of the example response section not being ridiculously long (at least until we have scrolling in that section of the docs) can you limit this response to maybe just the simplest metaschema? I think that would be either the "Open-ended Registration" or "OSF-Standard Pre-Data Collection Registration" metaschema.

File: swagger-spec/metaschemas/definition.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  type:
+    type: string
+    readOnly: true
+    description: 'The type identifier of the metaschema entity (`metaschemas`).'
+
+  attributes:
+    type: object
+    title: Attributes
+    readOnly: false
```
Comment (caseyrollins) [inline swagger-spec/metaschemas/definition.yaml]: Attributes should be `readOnly: true`

File: swagger-spec/metaschemas/definition.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    description: 'The properties of the metaschema entity.'
+    properties:
+      name:
+        type: string
+        readOnly: true
+        description: 'Name of the registration schema'
```
Comment (caseyrollins) [inline swagger-spec/metaschemas/definition.yaml]: "The name of the metaschema."

File: swagger-spec/metaschemas/definition.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      schema_version:
+        type: integer
+        readOnly: true
+        description: 'Latest version of the schema'
```
Comment (caseyrollins) [inline swagger-spec/metaschemas/definition.yaml]: "The latest version of the metaschema."

File: swagger-spec/metaschemas/definition.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      schema:
+        # TODO: Swagger UI seems to lack support for dictionaries.
+        # This property should not show as object
+        type: object
+        additionalProperties:
+          type: array
+          items:
+            type: string
+        readOnly: true
+        description: 'Registration schema details'
```
Comment (caseyrollins) [inline swagger-spec/metaschemas/definition.yaml]: "The details of the metaschema which contain the supplemental questions to accompany a registration."

File: swagger-spec/metaschemas/definition.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+  links:
+    type: object
+    title: Links
+    readOnly: true
+    description: 'URLs to alternative representations of the user entity.'
```
Comment (caseyrollins) [inline swagger-spec/metaschemas/definition.yaml]: "URLs to alternative representations of the metaschema entity."

File: swagger-spec/metaschemas/definition.yaml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        description: 'Name of the registration schema'
+      schema_version:
+        type: integer
+        readOnly: true
+        description: 'Latest version of the schema'
+      schema:
+        # TODO: Swagger UI seems to lack support for dictionaries.
+        # This property should not show as object
+        type: object
+        additionalProperties:
```
Comment (caseyrollins) [inline swagger-spec/metaschemas/definition.yaml]: What is the `additionalProperties` key here specifying?

File: swagger-spec/metaschemas/definition.yaml [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        readOnly: true
+        description: 'Name of the registration schema'
+      schema_version:
+        type: integer
+        readOnly: true
+        description: 'Latest version of the schema'
+      schema:
+        # TODO: Swagger UI seems to lack support for dictionaries.
+        # This property should not show as object
+        type: object
```
Comment (caseyrollins) [inline swagger-spec/metaschemas/definition.yaml]: Object means the same thing in this context, so this is fine 👍

File: swagger-spec/swagger.yaml [hunk 9]
```diff
@@ -294,3 +294,13 @@ paths:
 
   /wikis/{wiki_id}/content/:
     $ref: 'wikis/content.yaml'
+
+
+  ###################
+  #   METASCHEMAS   #
```
Comment (caseyrollins) [inline swagger-spec/swagger.yaml]: Can you make sure that this file stays in alphabetical order? The metaschemas URLs should be between institutions and nodes here.

**Review by caseyrollins — CHANGES_REQUESTED** [review]
A few questions and changes needed, but this is looking great :octocat:

File: swagger-spec/metaschemas/definition.yaml (hunk 2 above)
Comment (fabmiz) [reply swagger-spec/metaschemas/definition.yaml]: Ok. Got it!

File: swagger-spec/metaschemas/definition.yaml (hunk 3 above)
Comment (fabmiz) [reply swagger-spec/metaschemas/definition.yaml]: Done.

File: swagger-spec/metaschemas/definition.yaml (hunk 4 above)
Comment (fabmiz) [reply swagger-spec/metaschemas/definition.yaml]: Done.

File: swagger-spec/metaschemas/definition.yaml (hunk 7 above)
Comment (fabmiz) [reply swagger-spec/metaschemas/definition.yaml]: Swagger UI does not seem to have a way to represent dictionaries. So this defines a dictionary where the values are arrays of strings.

File: swagger-spec/metaschemas/list.yaml (hunk 1 above)
Comment (fabmiz) [reply swagger-spec/metaschemas/list.yaml:29]: Go it. Done.

**Review by caseyrollins — APPROVED** [review]
💯

**Pull request merged**